BetterPasswordPlugin.inc.php

```diff
@@ -160,24 +160,52 @@ function checkPassword($hookName, $args) {
+                        $blacklistFileSetting = $this->getSetting(CONTEXT_SITE, 'betterPasswordBlacklistFiles');
+                        if (is_null($blacklistFileSetting)) {
+                            $blacklistFileSettingHash = array();
+                            foreach ($this->getBlacklists() as $filename) {
+                                $blacklistFileSettingHash[$filename] = sha1_file($filename);
+                            }
+                            $this->updateSetting(CONTEXT_SITE, 'betterPasswordBlacklistFiles', $blacklistFileSettingHash);
+                        }
+                        else {
+                            $currBlacklist = array();
+                            foreach ($this->getBlacklists() as $filename) {
+                                $currBlacklist[$filename] = sha1_file($filename);
+                            }
+                            $prevBlacklist = $this->getSetting(CONTEXT_SITE, 'betterPasswordBlacklistFiles');
+                            $t = empty(array_udiff_uassoc($currBlacklist, $prevBlacklist, "strcmp", "strcmp"));
+                            $s = empty(array_udiff_uassoc($prevBlacklist, $currBlacklist, "strcmp", "strcmp"));
+                            
+                            if (!empty(array_udiff_uassoc($currBlacklist, $prevBlacklist, "strcmp", "strcmp")) || !empty(array_udiff_uassoc($prevBlacklist, $currBlacklist, "strcmp", "strcmp"))) {
+                                $tempFilename = $this->handleTempFile();
+                                $fpTemp = fopen($tempFilename, 'w');
+                                foreach ($this->getBlacklists() as $filename) {
+                                    $fpPass = fopen($filename, "r");
+                                    if (flock($fpTemp, LOCK_EX)) {
+                                        while (!feof($fpPass)) {
+                                            $passwordLine = fgets($fpPass);
+                                            if (strlen($passwordLine) >= $minLengthPass) {
+                                                fwrite($fpTemp, $passwordLine);
+                                            }
+                                        }
+                                        flock($fpTemp, LOCK_UN);
+                                    }
+                                    fclose($fpPass);
+                                }
+                                fclose($fpTemp);
+                                $flushCache = CacheManager::getManager()->flush('badPasswords');
+                                $this->updateSetting(CONTEXT_SITE, 'betterPasswordBlacklistFiles', $currBlacklist);   
+                            }
+                        }
+                        $cache = CacheManager::getManager()->getCache('badPasswords', $password_hash, array($this, '_PasswordCacheMiss'));
+                        $bad_password = $cache->get($sha_pass);
```

Comment:
The introduction of `$bad_password` when we already have a variable `$badPassword` is confusing.  Select a better name.